src/lib/math/pcurves/pcurves_impl/pcurves_wrap.h

```diff
@@ -12,6 +12,18 @@
 
 namespace Botan::PCurve {
 
+template <typename C, typename = int>
+struct CurveHasFeInvert2 : std::false_type {};
+
+template <typename C>
+struct CurveHasFeInvert2<C, decltype(&C::fe_invert2, 0)> : std::true_type {};
+
+template <typename C, typename = int>
+struct CurveHasScalarInvert : std::false_type {};
+
+template <typename C>
+struct CurveHasScalarInvert<C, decltype(&C::scalar_invert, 0)> : std::true_type {};
```

Comment:
Just as [in the other pull request](https://github.com/randombit/botan/pull/4209#discussion_r1677433043); let's make this a concept:

```C++
template <typename C>
concept supports_fast_scalar_inversion = requires(const typename C::Scalar& fe) {
   { C::scalar_invert(fe) } -> std::same_as<typename C::Scalar>;
};
```